sys/net/gnrc/network_layer/ipv6/gnrc_ipv6.c

```diff
@@ -888,11 +890,19 @@ static void _receive(gnrc_pktsnip_t *pkt)
     if (byteorder_ntohs(hdr->len) < pkt->size) {
         gnrc_pktbuf_realloc_data(pkt, byteorder_ntohs(hdr->len));
     }
+#ifdef MODULE_GNRC_SIXLOWPAN 
+    else if (byteorder_ntohs(hdr->len) >
+             (gnrc_pkt_len_upto(pkt, GNRC_NETTYPE_IPV6))) {
```

Comment:
If the node have both 6LoWPAN and IPv6 interface, this condition may not catch invalid packets.

Running debugger on `tests/gnrc_ipv6_ext` with 6LoWPAN enabled (i.e. adding `USEMODULE += gnrc_sixlowpan_default` to the Makefile) shows `gnrc_pkt_len_upto(pkt, GNRC_NETTYPE_IPV6) == 82` while  `byteorder_ntohs(hdr->len) == 42` because the former includes IPv6 header. So, a malicious packet with `byteorder_ntohs(hdr->len) == 82` will pass this validation even if its actual length is 42.